x11-libs/qwt/qwt-6.2.0-r2.ebuild

```diff
@@ -39,71 +50,91 @@ RESTRICT="test"
+			VERSION = ${PV/_*}
+			QWT_VER_MAJ = $(ver_cut 1)
+			QWT_VER_MIN = $(ver_cut 2)
+			QWT_VER_PAT = $(ver_cut 3)
+			QWT_VERSION = ${PV/_*}
+		EOF
+
+		use designer && echo "QWT_CONFIG += QwtDesigner" >> qwtconfig.pri
+		use opengl && echo "QWT_CONFIG += QwtOpenGL" >> qwtconfig.pri
+		use polar && echo "QWT_CONFIG += QwtPolar" >> qwtconfig.pri
+		use svg && echo "QWT_CONFIG += QwtSvg" >> qwtconfig.pri
 
-	cat > qwtconfig.pri <<-EOF || die
-		QWT_INSTALL_LIBS = "${EPREFIX}/usr/$(get_libdir)"
-		QWT_INSTALL_HEADERS = "${EPREFIX}/usr/include/qwt6"
-		QWT_INSTALL_DOCS = "${EPREFIX}/usr/share/doc/${PF}"
-		QWT_CONFIG += QwtPlot QwtWidgets QwtPkgConfig
-		VER_MAJ = $(ver_cut 1)
-		VER_MIN = $(ver_cut 2)
-		VER_PAT = $(ver_cut 3)
-		VERSION = ${PV/_*}
-		QWT_VER_MAJ = $(ver_cut 1)
-		QWT_VER_MIN = $(ver_cut 2)
-		QWT_VER_PAT = $(ver_cut 3)
-		QWT_VERSION = ${PV/_*}
-	EOF
-
-	use designer && echo "QWT_CONFIG += QwtDesigner" >> qwtconfig.pri
-	use opengl && echo "QWT_CONFIG += QwtOpenGL" >> qwtconfig.pri
-	use polar && echo "QWT_CONFIG += QwtPolar" >> qwtconfig.pri
-	use svg && echo "QWT_CONFIG += QwtSvg" >> qwtconfig.pri
-
-	cat > qwtbuild.pri <<-EOF || die
-		QWT_CONFIG += qt warn_on thread release no_keywords
-		DEFINES += QWT_MOC_INCLUDE=1
-	EOF
-
-	echo "QWT_CONFIG += QwtDll" >> qwtconfig.pri
-
-	cat >> qwtconfig.pri <<-EOF || die
-		QWT_INSTALL_PLUGINS   = "${EPREFIX}$(qt5_get_plugindir)/designer"
-		QWT_INSTALL_FEATURES  = "${EPREFIX}$(qt5_get_mkspecsdir)/features"
-	EOF
-	sed \
-		-e 's/target doc/target/' \
-		-e "/^TARGET/s:(qwt):(qwt6-qt5):g" \
-		-e "/^TARGET/s:qwt):qwt6-qt5):g" \
-		-e "s:QWT_SONAME=libqwt.so:QWT_SONAME=libqwt6-qt5.so:g" \
-		-i src/src.pro || die
-
-	sed \
-		-e '/qwtAddLibrary/s:(qwt):(qwt6-qt5):g' \
-		-e '/qwtAddLibrary/s:qwt):qwt6-qt5):g' \
-		-i qwt.prf designer/designer.pro examples/examples.pri || die
+		cat > qwtbuild.pri <<-EOF || die
+			QWT_CONFIG += qt warn_on thread release no_keywords
+			DEFINES += QWT_MOC_INCLUDE=1
+		EOF
+
+		echo "QWT_CONFIG += QwtDll" >> qwtconfig.pri
+
+		cat >> qwtconfig.pri <<-EOF || die
+			QWT_INSTALL_PLUGINS   = "${EPREFIX}/usr/$(get_libdir)/${MULTIBUILD_VARIANT}/plugins/designer"
+			QWT_INSTALL_FEATURES  = "${EPREFIX}/usr/$(get_libdir)/${MULTIBUILD_VARIANT}/mkspecs/features"
+		EOF
+		sed \
+			-e 's/target doc/target/' \
+			-e "/^TARGET/s:(qwt):(qwt6-${MULTIBUILD_VARIANT}):g" \
+			-e "/^TARGET/s:qwt):qwt6-${MULTIBUILD_VARIANT}):g" \
+			-e "s:QWT_SONAME=libqwt.so:QWT_SONAME=libqwt6-${MULTIBUILD_VARIANT}.so:g" \
+			-i src/src.pro || die
+
+		sed \
+			-e "/qwtAddLibrary/s:(qwt):(qwt6-${MULTIBUILD_VARIANT}):g" \
+			-e "/qwtAddLibrary/s:qwt):qwt6-${MULTIBUILD_VARIANT}):g" \
+			-i qwt.prf designer/designer.pro examples/examples.pri || die
+	}
+	default
+	multibuild_copy_sources
+	multibuild_foreach_variant run_in_build_dir my_src_prepare
 }
 
 src_configure() {
-	eqmake5
+	my_src_configure() {
+		case ${MULTIBUILD_VARIANT} in
+			qt5) eqmake5 ;;
+			qt6) qmake6 ;;
```

Comment:
I wonder if we shouldn't wait until `qmake-utils.eclass` gains a `eqmake6` function. (CC @gentoo/qt )

In any case, as it is now, `qmake6` is an external function and therefore would need a `|| die`

Reply:
sure, let's wait for https://github.com/gentoo/gentoo/pull/31254 to include `eqmake6` ...